File: 2145-grid-game/grid-game.java
class Solution {
    public long gridGame(int[][] g) {
        long ans = Long.MAX_VALUE ;
        long pref = 0 ;
        long suf = 0 ;
        for(int i : g[0])
            suf += i ;
        for(int i = 0 ; i < g[0].length ; i ++){
            suf -= g[0][i];
            ans = Math.min(ans, Math.max(suf, pref)) ;
            pref += g[1][i] ;
        }
        return ans ;
    }
}

File: 2145-grid-game/grid-game.py
class Solution:
    def gridGame(self, g: List[List[int]]) -> int:
        ans = float('inf')
        pref, suf = 0, sum(g[0])
        for i in range(len(g[0])):
            suf -= g[0][i]
            ans = min(ans, max(suf, pref))
            pref += g[1][i] 
        return ans